src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TransitiveDependencyExtensions.cs

```diff
@@ -65,7 +65,13 @@ private static bool ContainsDependency(this IProject project, TargetFrameworkMon
 
         private static IEnumerable<LockFileTargetLibrary> GetAllDependencies(this IProject project, TargetFrameworkMoniker tfm)
         {
+            if (string.IsNullOrEmpty(project.LockFilePath))
```

Comment:
If you merge from main, you'll get a more robust check behind `project.NuGetReferences.GetAllDependencies()`

Reply:
I'll remove this. It was only added because the project.LockFilePath was null when unit testing and I couldn't easily mock the response from this method.